### ISSUE TITLE
Bad 1.21+ file structure?

### ISSUE THREAD

**Issue by mgcrea** [PaulStoffregen/cores#64]
While trying to make [Arduino-Makefile](https://github.com/sudar/Arduino-Makefile) work with the latest Arduino.app + Teensyduino 1.21, I've encountered a few issues and shared my progress on [this issue](https://github.com/sudar/Arduino-Makefile/issues/325). 

One of the contributors pointed out that the new directory layout used by teensyduino was odd:
>
teensyduino 1.21 (for 1.6.1) seems to use some very odd directory structure and boards.txt, doesn't even seem to match other 1.5+ implementations, and so doesn't work with the makefile.

Not sure what is the reference/"good" file layout/structure but was interested by your input on this issue.

**Comment by PaulStoffregen** [comment]
Yes, Teensy makes much more use of the parameter substitution from boards.txt into platform.txt.  If you're properly implementing Arduino's algorithms, platform.txt will resolve to the correct commands.  But if you're hard-coding to the specific recipes Arduino publishes, then of course Teensy's very different recipes will seem rather foreign.

Three custom Tools menus are also used to define several important settings.  You'll probably need to implement some way to select those 3 settings.  The available settings are different for each board.  You should probably also be aware I'm planning to add 3 more USB types later this year, with amazing USB functionality that's never been seen on any Arduino compatible board.

The Teensyduino installer also adds a couple minor patches to the IDE.

1: A recipe called "recipe.elfpatch.pattern" which is run if boards.txt defines "build.elfpatch".  This is only needed on the 2.0 board when Disk(internal) is used.  If you don't implement this, everything other than that 1 special USB option should work.

2: A special "{extra.time.local}" symbol is added to the build settings, which contains the "time_t" time the build was run.  You'll need to substitute an integer for this name.  If you use a fixed value, the only consequence is the optional RTC on Teensy 3.1 will not automatically set upon the first upload after a RTC battery removal.
